testflinger_agent/agent.py

```diff
@@ -165,10 +165,6 @@ def process_jobs(self):
                     # In this case, we need to mark the device offline
                     if exitcode == 46:
                         self.mark_device_offline()
-                        self.client.repost_job(job_data)
-                        shutil.rmtree(rundir)
-                        # Return NOW so we don't keep trying to process jobs
-                        return
```

Comment:
Don't we want to exit from the function here (return)?

Reply:
No, I'd actually like it to fall through and finish out submitting the results it has so far, which may be useful for debugging. That way the requester will see the message that it failed to provision and any other info gathered so far.